### ISSUE TITLE
Failure when requiring 'active_support/core_ext/date_time/compatibility'

### ISSUE THREAD

**Issue by kamaradclimber** [rails/rails#28918]
### Steps to reproduce

```
gem install activesupport -v 5.1.0
irb
require 'active_support/core_ext/date_time/compatibility'
```

### Expected behavior

require operation should return true.

### Actual behavior

```
NoMethodError: undefined method `remove_possible_method' for DateTime:Class
Did you mean?  remove_method
from /home/g_seux/.gem/ruby/2.4.0/gems/activesupport-5.1.0/lib/active_support/core_ext/date_time/compatibility.rb:6:in `<class:DateTime>'
```

### System configuration
**Rails version**: 5.1.0

**Ruby version**: 2.4.0

**Comment by mgreensmith** [comment]
I'm experiencing this too. Looks like `date_time/compatibility.rb` needs to require `active_support/core_ext/module/remove_method`.

**Comment by pixeltrix** [comment]
Okay, it should probably require that file - I'm just wondering why you're including that file by itself? If you want to use Active Support's time extensions you should require 'active_support/time' and not mix and match the files individually

**Comment by mgreensmith** [comment]
This actually breaks for me when requiring all of `active_support/core_ext` in an app running inside a docker container. Apparently the container's `aufs` filesystem globs directories in a different order than other common filesystems. In docker, `core_ext.rb` tries to require `date_time.rb` before the other classes that also require `module/remove_method`, exposing the missing require.

So it's more than just an isolated bug when requiring the single file in isolation.

(This was a real yak shave to debug.)

**Comment by rafaelfranca** [comment]
Does it also break if you require `active_support` before requiring that file?

**Comment by rafaelfranca** [comment]
Yup. Just confirmed

**Comment by kamaradclimber** [comment]
Thanks for the quick patch!

Le sam. 29 avr. 2017 00:01, Rafael França <notifications@github.com> a
écrit :

**Comment by faustoct** [comment]
so what's the patch? Any idea when the patch goes to stable release in 'bundle update'?

**Comment by rafaelfranca** [comment]
The patch is linked in this issue. It will go to the stable release in some weeks.

**Comment by domanhtien2011** [comment]
Currently, I am facing this issue. The rails version for my app is 
`gem 'rails', '~> 5.0.0', '>= 5.0.0.1'`
And I am using ruby 2.3.3. How can I apply this patch to fix the issue?

**Comment by wjordan** [comment]
Note that the original PR introducing this regression (#28147) was also backported to the `5-0-stable` (#28448) and `4-2-stable` (#28465) branches, but the fix in #28920 has only been backported to the `5-1-stable` branch (included in `5.1.1` and up).

A `4.2.x` branch with the backported regression was never released, but all current `5.0.x` releases have this issue.

**Comment by pixeltrix** [comment]
@wjordan I've backported them in 7fe69eaef4bf93418733953b8afaef1a757560e3 (4-2-stable) and
01dc2b5fdc6367fa1d3b7dd7aeaea4d1407fafbb (5-0-stable) though it's unlikely there will ever be a gem release, sorry.
